### PR TITLE
Allow Protocol to be Set for API Hosts

### PR DESCRIPTION
**Pull request by dubeejw**
- Allow different protocols to be used with the API hosts

### CODE REVIEW

**Comment by dubeejw** [comment]
PG1 447

**Comment by mdeuser** [comment]
This PR is blocking mainBluewhisk builds.
https://bluemix.pagerduty.com/incidents/PQRSGUP

**Comment by dubeejw** [comment]
PG1 447.

File: tools/go-cli/go-whisk-cli/commands/util.go [hunk 1]
```diff
@@ -822,3 +823,14 @@ func checkArgs(args []string, minimumArgNumber int, maximumArgNumber int, comman
     }
 }
 
+func urlBase() (*url.URL, error)  {
+    urlBase := fmt.Sprintf("%s/api/", Properties.APIHost)
+    url, err := url.Parse(urlBase)
+
+    if len(url.Scheme) == 0 || len(url.Host) == 0 {
```
Comment (markusthoemmes) [inline tools/go-cli/go-whisk-cli/commands/util.go]: Why `url.Host` also == 0?

Comment (dubeejw) [inline tools/go-cli/go-whisk-cli/commands/util.go]: Apparently Go puts the host in the Scheme field if no protocol is provided.

**Comment by dubeejw** [comment]
PG2 154. This fixes an alert.

**Pull request merged**